code/lighting/lighting.cpp

```diff
@@ -122,22 +122,52 @@ static auto DeferredLightingOption = options::OptionBuilder<bool>("Graphics.Defe
                   .parser(parse_deferred_lighting_func)
                   .finish();
 
+static bool DeferredCockpitLightingEnabled = true;
 
+static void parse_deferredcockpit_lighting_func()
+{
+	bool enabled;
+	stuff_boolean(&enabled);
+	DeferredCockpitLightingEnabled = enabled;
+}
+
+static auto DeferredCockpitLightingOption = options::OptionBuilder<bool>("Graphics.DeferredCockpitLighting",
+                  std::pair<const char*, int>{"Deferred Cockpit Lighting", 1864},
+                  std::pair<const char*, int>{"Enables or disables deferred lighting in cockpits", 1865})
+                  .category(std::make_pair("Graphics", 1825))
+                  .default_func([]() { return DeferredCockpitLightingEnabled;})
+                  .level(options::ExpertLevel::Advanced)
+                  .bind_to_once(&DeferredCockpitLightingEnabled)
+                  .importance(60)
+                  .parser(parse_deferredcockpit_lighting_func)
+                  .finish();
+
+
+// These deferred enabled functions used to get the value of the option object itself,
+// however that is not a free operation and changing it requires a restart anyway.
+// If the restart requirement is lifted, then care should be taken to cache this value
+// and never look it up more than once a frame;
+// otherwise the performance footprint is measurable enough to worry about.
```

Comment:
This comment would appear to also apply to the new cockpit deferred lighting option, yeah? If so I might add a short comment to the new function saying to reference this comment so there's no confusion and it's less likely to get missed if the restart requirement is ever changed for either function.

Reply:
Good point, just updated, thanks!